mnekrasov/files-file-systems/reverse-copying/main.c

```diff
@@ -0,0 +1,187 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <errno.h>
+
+#define SUCCESS 0
+#define ERROR -1
+#define EQUAL 0
+
+void reverseString(char *string) {
+    int stringLen = strlen(string);
+
+    for (int i = 0; i < stringLen / 2; ++i) {
+        char temp = string[i];
+        string[i] = string[stringLen - 1 - i];
+        string[stringLen - 1 - i] = temp;
+    }
+}
+
+void reverseFile(const char *srcFilePath, const char *dstFilePath) {
+    FILE *srcFile = fopen(srcFilePath, "rb");
+    if (!srcFile) {
+        perror("Impossible to open source file for reading");
+        return;
+    }
+
+    fseek(srcFile, 0, SEEK_END);
+    long srcFileSize = ftell(srcFile);
```

Comment:
У всех системных вызовов (не только у ftell) нужно проверять возвращаемое значение, иначе выполнение уйдет в undefined behaviour 
> Upon successful completion, fgetpos(), fseek(), fsetpos() return 0, and ftell() returns the current offset.  Otherwise, -1 is returned and errno is set to indicate the error.